include/stream-client/detail/logger.ipp

```diff
@@ -0,0 +1,89 @@
+#pragma once
+
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+
+namespace stream_client {
+
+namespace logging {
+
+inline const log_interface* get_logger(const log_interface* new_logger = nullptr) {
```

Comment:
This looks like function for internal usage, better to put it in the `detail` namespace (`logging::detail`)